app/knights/battle.py

```diff
@@ -0,0 +1,37 @@
+from app.knights.knights import lancelot, mordred, arthur, red_knight
+from app.config.config import KNIGHTS
+
+
+# BATTLE:
+class Battle:
+    @staticmethod
+    def battle(knights_config: dict) -> dict:
+        # 1 Lancelot vs Mordred:
+        lancelot.hp -= mordred.power - lancelot.protection
+        mordred.hp -= lancelot.power - mordred.protection
+
+        # check if someone fell in battle
+        if lancelot.hp <= 0:
+            lancelot.hp = 0
+
+        if mordred.hp <= 0:
+            mordred.hp = 0
+
+        # 2 Arthur vs Red Knight:
+        arthur.hp -= red_knight.power - arthur.protection
```

Comment:
The expression `arthur.hp -= red_knight.power - arthur.protection` also assumes a direct `protection` attribute. Ensure that protection is calculated from the armour list.